### ISSUE TITLE
MyTested.AspNetCore not working when migrate to VS2017 and all xproj are converted to csproj

### ISSUE THREAD

**Issue by GeorgiNik** [ivaylokenov/MyTested.AspNetCore.Mvc#275]
I started to use Visual studio 2017 in order to work all xproj were converted to csproj. My test project stoped working when you run tests the following exeption start to rise when MyController<T>.Instace() was executed:

`Error Message:
 System.TypeInitializationException : The type initializer for 'MyTested.AspNetCore.Mvc.MyController`1' threw an exception.
---- System.ArgumentNullException : Value cannot be null.
Parameter name: assembly
Stack Trace:
   at MyTested.AspNetCore.Mvc.MyController`1.Instance()
   at RecommendIT.Projects.Tests.Controllers.ProjectsControllerTest.Delete_ShouldReturnNotFound_WhenProjectDontExist(String id) in C:\Users\georgenik\Desktop\RIT3\Projects\RecommendIT.Projects.Tests\Controllers\ProjectsControllerTest.cs:line 151
----- Inner Stack Trace -----
   at Microsoft.Extensions.DependencyModel.DependencyContextLoader.Load(Assembly assembly)
   at Microsoft.Extensions.DependencyModel.DependencyContext.LoadDefault()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at MyTested.AspNetCore.Mvc.Internal.Application.TestApplication.LoadDependencyContext()
   at MyTested.AspNetCore.Mvc.Internal.Application.TestApplication.Initialize()
   at MyTested.AspNetCore.Mvc.Internal.Application.TestApplication.TryInitialize()
   at MyTested.AspNetCore.Mvc.MyController`1..cctor()
Failed   RecommendIT.Projects.Tests.Controllers.ProjectsControllerTest.Create_ShouldReturnExistingProjectError_WhenProjectNameExists(entity: ProjectModel { CurrentUserId = "CurrentUserId5963b53a-d77f-447b-bf3b-d5b12e4ab955", CurrentUserName = "CurrentUserNameca7264a4-bc1f-4b67-874a-3c5af4ad66a"..., Description = "Description75485bb7-38a2-453e-a8a1-75452127f63d", DueDate = 2018-10-19T11:27:45.0982800, EscalationPointId = "EscalationPointId48fbd17c-3864-4689-b7c5-738d6f4ab"..., ... `})

This is my csproj file: 
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
    <PreserveCompilationContext>true</PreserveCompilationContext>
    <AssemblyName>RecommendIT.Projects.Tests</AssemblyName>
    <PackageId>RecommendIT.Projects.Tests</PackageId>
    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
    <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
    <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
    <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
  </PropertyGroup>

  <ItemGroup>
    <None Remove="appsettings.json" />
  </ItemGroup>

  <ItemGroup>
    <Content Include="appsettings.json">
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </Content>
  </ItemGroup>

  <ItemGroup>
    <None Update="appsettings.json;testconfig.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\Common\RecommendIT.Common.Data\RecommendIT.Common.Data.csproj" />
    <ProjectReference Include="..\RecommendIT.Projects.ProjectsWebApi\RecommendIT.Projects.ProjectsWebApi.csproj" />
    <ProjectReference Include="..\RecommendIT.Projects.BusinessModels\RecommendIT.Projects.BusinessModels.csproj" />
    <ProjectReference Include="..\RecommendIT.Projects.Data\RecommendIT.Projects.Data.csproj" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
    <PackageReference Include="Microsoft.TestPlatform.TestHost" Version="15.0.0" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
    <PackageReference Include="FluentAssertions" Version="4.19.2" />
    <PackageReference Include="Moq" Version="4.7.12" />
    <PackageReference Include="MyTested.AspNetCore.Mvc.Universe" Version="1.1.0" />
    <PackageReference Include="xunit" Version="2.2.0" />
    <PackageReference Include="Microsoft.DotNet.InternalAbstractions" Version="1.0.0" />
    <PackageReference Include="AutoFixture" Version="3.50.2" />
    <PackageReference Include="AutoFixture.AutoMoq" Version="3.50.2" />
    <PackageReference Include="AutoFixture.Xunit2" Version="3.50.2" />
    <PackageReference Include="MongoDB.Driver" Version="2.4.3" />
    <PackageReference Include="Microsoft.Extensions.Options" Version="1.1.1" />
    <PackageReference Include="RecommendIT.Common.Data" Version="1.0.0-*" />
  </ItemGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
    <Reference Include="System" />
    <Reference Include="Microsoft.CSharp" />
  </ItemGroup>

</Project>
```

@ivaylokenov  do you have any suggestions what is causing that problem?

**Comment by kuskmen** [comment]
Same here, I guess 1.1.0 is not compatible with VS2017, but it would be nice to have support anytime soon.

**Comment by ivaylokenov** [comment]
@GeorgiNik @kuskmen 

Hi guys,

The library is now ported to .NET Core 2 and this issue is resolved. I will try to explain why it was delayed by so much time.

Back in March 2017 when .NET Core migrated to the new (old) .csproj file format, I was having difficulties with the project as it can be seen here - #270 (comment)

Visual Studio was continually crashing, tests did not run correctly, and the overall developing experience was terrible. I tried for 2-3 months to port the library but could not succeed.

During that time, I started developing severe health issues from staying in front of my computer too much. I was working 12 to 16 hours from Monday to Sunday with not many breaks for 2-3 years straight. I had to stop doing it for an indefinite time. My GitHub activity shows this pretty clearly: https://github.com/ivaylokenov

The truth is - I like this testing library and have much fun developing it, so I was confused and did not know what to tell people about its future maintenance. I did not want to abandon it forever but did not know when I would be able to work actively again on it. I now realize that it was a mistake to be silent all this time and should have put an explanation earlier. The fact that my personal life was very stressful for the last two years is not an excuse.

Back in October 2018, I opened the code and started working actively on bringing back the library to life. It took me 6 months because I added lots of new features like third-party DI support, all possible action results, every single action filter, lots of helper methods for easier testing, more straightforward configuration, performance fixes and more. Version 2.2 is available on NuGet already, and version 2.1 will be published in the next few days. I will then update the tutorial, documentation, the web site, and will start working on .NET Core 3.

I am sorry for the inconvenience caused by this library and me. I truly am.

As compensation (if that is even possible), all versions before 3.0 do not have any limitations and can be used without any license code.

The good news is that I found a small private investor who is sponsoring the project. I can actively put 20 to 30 hours per week in the development of MyTested.AspNetCore.Mvc. If I manage to increase the usage and find clients, the investor will increase the budget and I will be able to hire another developer to help me. Additionally, I will put a sponsorship program with logos, and hopefully, I will be able to maintain the project steadily in the years to come.

Thank you for reading and understanding.

Ivaylo
